src/schema/v2/artworksForUser.ts

```diff
@@ -0,0 +1,129 @@
+import { GraphQLString, GraphQLFieldConfig, GraphQLInt } from "graphql"
+import { connectionFromArraySlice } from "graphql-relay"
+import { convertConnectionArgsToGravityArgs, extractNodes } from "lib/helpers"
+import { CursorPageable, pageable } from "relay-cursor-paging"
+import { createPageCursors } from "schema/v1/fields/pagination"
+import { ResolverContext } from "types/graphql"
+import { artworkConnection } from "schema/v2/artwork"
+import gql from "lib/gql"
+
+const MAX_ARTISTS = 50
+const MAX_ARTWORKS = 100
+const MIN_AFFINITY_SCORE = 0.5
+
+export const artworksForUser: GraphQLFieldConfig<void, ResolverContext> = {
+  description: "A connection of artworks for a user.",
+  type: artworkConnection.connectionType,
+  args: pageable({
+    page: { type: GraphQLInt },
+    userId: { type: GraphQLString },
+  }),
+  resolve: async (
+    _root,
+    args: CursorPageable,
+    {
+      vortexGraphqlLoader,
+      vortexGraphqlLoaderFactory,
+      artworksLoader,
+      appToken,
+    }
+  ) => {
+    if (!vortexGraphqlLoader || !artworksLoader) {
+      const vortexGraphqlLoader = vortexGraphqlLoaderFactory(appToken)
+
+      const userId = `"${args.userId}"`
+
+      const vortexResult = await vortexGraphqlLoader({
+        query: gql`
+          query artistAffinitiesQuery {
+            artistAffinities(userId: ${userId}, first: ${MAX_ARTISTS}, minScore: ${MIN_AFFINITY_SCORE}) {
+              totalCount
+              edges {
+                node {
+                  artistId
+                  score
+                }
+              }
+            }
+          }
+        `,
+      })()
+
+      const artistIds = extractNodes(vortexResult.data?.artistAffinities).map(
+        (node: any) => node?.artistId
+      )
+
+      let artworks = []
+
+      const { page, size, offset } = convertConnectionArgsToGravityArgs(args)
+
+      if (artistIds?.length) {
+        artworks = await artworksLoader({
+          artist_ids: artistIds,
+          sort: "-published_at",
+          availability: "for sale",
+          size,
+          offset,
+        })
+      }
+
+      // TODO: get count from artworks loader to optimize pagination
+      const count = artworks.length === 0 ? 0 : MAX_ARTWORKS
+
+      return {
+        totalCount: count,
+        pageCursors: createPageCursors({ ...args, page, size }, count),
+        ...connectionFromArraySlice(artworks, args, {
```

Comment:
all of this connection stuff can be simplified using our connection helper patterns: https://github.com/artsy/metaphysics/blob/main/docs/pagination.md

Reply:
I just copy/pasted this part from the original query here:

https://github.com/artsy/metaphysics/blob/main/src/schema/v2/me/newWorksByInterestingArtists.ts#L70-L78

Reply:
We should update to use pagination helpers! it will make this code easier to read and simpler and in line with new patterns.

Reply:
Is this what you had in mind @damassi? [857af3e](https://github.com/artsy/metaphysics/pull/4030/commits/857af3ec288275b1b96aa9f175efd0df0082096c)